### DISCUSSION TITLE
[Help] How to work with different build targets

### DISCUSSION THREAD

**Discussion by akshaim** [rizsotto/Bear#507] [Q&A] [answered]
I am currently working on an embedded system  RTOS project that has multiple build targets ( basically different hardware boards and sensors). I would pass the build parameters to the make build system and then the build system includes the right libraries ( src and headers) based on the build target. 

Essentially what I do if I want to compile for a particular board I would use `make BOARD=Nucleofoo` likewise for another board `make BOARD=Nucleobar`. I would like to CPPCheck on the code and right now I am using bear to generate the compile_commands.json. Unfortunately this has to be done for each target recursively. Is there a better way of doing this ? So that I can compile all the dependencies to a single JSON ?

**Comment by rizsotto** [comment, marked as answer]
Hey @akshaim , this is an interesting use case. I think the way you currently doing this seems to be most safe way. If you want to speed this up, you probably need to use different build directories, so you can run the builds and the check parallel to each other. It's easy to do this way in CI environments. But for local development, you might want to write a shell/python script to do it for you.

This is less effort than trying to merge the different compilation databases into one.

**Comment by akshaim** [comment]
Cool. Thanks for the answer
